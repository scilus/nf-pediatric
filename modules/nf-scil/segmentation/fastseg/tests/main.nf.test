nextflow_process {

    name "Test Process SEGMENTATION_FASTSEG"
    script "../main.nf"
    process "SEGMENTATION_FASTSEG"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/fastseg"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "heavy.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("segmentation - fastseg") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/anat/anat_image.nii.gz", checkIfExists: true)
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
