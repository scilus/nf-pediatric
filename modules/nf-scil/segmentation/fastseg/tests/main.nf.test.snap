{
    "segmentation - fastseg": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_wm.nii.gz:md5,f849e268007a747714e342162f13f631"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_gm.nii.gz:md5,39396421c6d1d32ea7943a5ec7a49dce"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_csf.nii.gz:md5,d0fbc088059f17faff38fbcd33257d27"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_wm.nii.gz:md5,1d5a9655d4e3c832b14c83e9424b00fc"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_gm.nii.gz:md5,f7e51eb14af645c687db5d2c6a4f5bf4"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_csf.nii.gz:md5,9c14d8f70e69d641d9c214d0d52b3663"
                    ]
                ],
                "6": [
                    "versions.yml:md5,e9c2afb5537207544dc1b54cbdc389a4"
                ],
                "csf_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_csf.nii.gz:md5,9c14d8f70e69d641d9c214d0d52b3663"
                    ]
                ],
                "csf_mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_csf.nii.gz:md5,d0fbc088059f17faff38fbcd33257d27"
                    ]
                ],
                "gm_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_gm.nii.gz:md5,f7e51eb14af645c687db5d2c6a4f5bf4"
                    ]
                ],
                "gm_mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_gm.nii.gz:md5,39396421c6d1d32ea7943a5ec7a49dce"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e9c2afb5537207544dc1b54cbdc389a4"
                ],
                "wm_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__map_wm.nii.gz:md5,1d5a9655d4e3c832b14c83e9424b00fc"
                    ]
                ],
                "wm_mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mask_wm.nii.gz:md5,f849e268007a747714e342162f13f631"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T15:30:44.012472"
    }
}